src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs

```diff
@@ -296,54 +277,19 @@ public static IResourceBuilder<PostgresServerResource> WithPgWeb(this IResourceB
             {
                 // Add the bookmarks to the pgweb container
 
-                // Create a folder using IAspireStore. Its name is deterministic, based on all the database resources
-                // such that the same folder is reused across persistent usages, and changes in configuration require
-                // new folders.
-
                 var postgresInstances = builder.ApplicationBuilder.Resources.OfType<PostgresDatabaseResource>();
 
-                var aspireStore = e.Services.GetRequiredService<IAspireStore>();
-
-                var tempDir = WritePgWebBookmarks(postgresInstances, out var contentHash);
+                var bookmarkFiles = WritePgWebBookmarks(postgresInstances);
 
-                // Create a deterministic folder name based on the content hash such that the same folder is reused across
-                // persistent usages.
-                var pgwebBookmarks = Path.Combine(aspireStore.BasePath, $"{pgwebContainer.Name}.{Convert.ToHexString(contentHash)[..12].ToLowerInvariant()}");
-
-                try
+                pgwebContainerBuilder.WithCreateFile("/.pgweb", new()
                 {
-                    Directory.CreateDirectory(pgwebBookmarks);
-
-                    // Grant listing access to the bookmarks folder on unix like systems.
-                    if (!OperatingSystem.IsWindows())
+                    new ContainerDirectory
                     {
-                        File.SetUnixFileMode(pgwebBookmarks, FileMode755);
-                    }
-
-                    foreach (var file in Directory.GetFiles(tempDir))
-                    {
-                        // Target is overwritten just in case the previous attempts has failed
-                        var destinationPath = Path.Combine(pgwebBookmarks, Path.GetFileName(file));
-                        File.Copy(file, destinationPath, overwrite: true);
-
-                        if (!OperatingSystem.IsWindows())
-                        {
-                            File.SetUnixFileMode(destinationPath, FileMode644);
-                        }
-                    }
-
-                    pgwebContainerBuilder.WithBindMount(pgwebBookmarks, "/.pgweb/bookmarks");
-                }
-                finally
-                {
-                    try
-                    {
-                        Directory.Delete(tempDir, true);
-                    }
-                    catch
-                    {
-                    }
-                }
+                        Name = "bookmarks",
```

Comment:
In `DirectoryInfo` it's named `path`.

Reply:
DirectoryInfo and FileInfo both seem to have a Name property that corresponds in purpose to this Name property.